crates/sui-transaction-builder/src/lib.rs

```diff
@@ -331,6 +331,8 @@ impl<Mode: ExecutionMode> TransactionBuilder<Mode> {
             } => ObjectArg::SharedObject {
                 id,
                 initial_shared_version,
+                // todo(RWLocks) - do we want to parametrise this?
+                mutable: true, // using mutable reference by default here.
```

Comment:
Eventually we should get the SDKs to fetch the package and infer this for us, but I can't think of a good way for them to efficiently get a single function signature today, so currently, the best path I see forward is to parameterize it, and have some way for users of the SDK to supply it.

Reply:
cc @666lcz , this will need to be replicated in the TS SDK (new parameter, defaulting to true for all shared object input parameters).

Reply:
Actually, I just spotted the note about default value in JSON-RPC, so this is less urgent, but it is worth doing sooner rather than later because eventually we will need to special case `Clock` to always be passed as immutable.